pkgs/applications/misc/inochi2d/generic.nix

```diff
@@ -0,0 +1,114 @@
+{ lib
+, buildDubPackage
+, fetchFromGitHub
+, writeShellScriptBin
+, makeDubDep
+
+, cmake
+, gettext
+, copyDesktopItems
+, makeDesktopItem
+, makeWrapper
+
+, dbus
+, freetype
+, SDL2
+, gnome
+}:
+
+let
+  cimgui-src = fetchFromGitHub {
+    owner = "Inochi2D";
+    repo = "cimgui";
+    rev = "49bb5ce65f7d5eeab7861d8ffd5aa2a58ca8f08c";
+    hash = "sha256-XcnZbIjwq7vmYBnMAs+cEpJL8HB8wrL098FXGxC+diA=";
+    fetchSubmodules = true;
+  };
```

Comment:
You mentioned this was included as a git submodule in the other projects. Is it not possible to simply use fetchSubmodules there aswell?

Reply:
`cimgui` is not a submodule of `inochi-session` or `inochi-creator`, but it's a submodule of `i2d-imgui`.
However `i2d-cimgui` is currently only used as a `dub` dependency, which doesn't know it's a git repo.

Upstream solves it in its builder by not using `dub` directly
https://github.com/Inochi2D/inochi-session/blob/755015c5c4a68cbe647ad4588955cbc13646d32a/.github/workflows/build-and-release.yml#L37

Or by manually fetching the submodules https://github.com/Inochi2D/i2d-imgui/blob/e34f8ba04c0085be7ee83a8df200cf2ffb30bfd3/deps/CMakeLists.txt#L7

I do it this way because this way the auto-generated `deps.nix` doesn't need to be changed.

Reply:
Could you add that info as a comment? For future maintainers, that's extremely valuable info to have.